pkg/conversion/mapi2capi/machine.go

```diff
@@ -17,113 +17,66 @@ limitations under the License.
-	if len(mapiNodeLabels) == 0 {
-		return field.ErrorList{}
-	}
-
-	if capiNodeLabels == nil {
-		capiNodeLabels = map[string]string{}
-	}
-
-	errs := field.ErrorList{}
-
-	// TODO(OCPCLOUD-2680/2897): Not all the labels on the CAPI Machine are propagated down to the corresponding CAPI Node, only the "CAPI Managed ones" are.
-	// These are those prefix by "node-role.kubernetes.io" or in the domains of "node-restriction.kubernetes.io" and "node.cluster.x-k8s.io".
-	// See: https://github.com/kubernetes-sigs/cluster-api/pull/7173
-	// and: https://github.com/fabriziopandini/cluster-api/blob/main/docs/proposals/20220927-label-sync-between-machine-and-nodes.md
-	for k, v := range mapiNodeLabels {
-		if !conversionutil.IsCAPIManagedLabel(k) {
-			errs = append(errs, field.Invalid(fldPath.Key(k), v, "label propagation is not currently supported for this label"))
-		}
-
-		capiNodeLabels[k] = v
-	}
-
-	return errs
-}
-
-// getCAPILifecycleHookAnnotations returns the annotations that should be added to a CAPI Machine to represent the lifecycle hooks.
-func getCAPILifecycleHookAnnotations(hooks mapiv1.LifecycleHooks) map[string]string {
-	annotations := make(map[string]string)
-
-	for _, hook := range hooks.PreDrain {
-		annotations[fmt.Sprintf("%s/%s", capiv1.PreDrainDeleteHookAnnotationPrefix, hook.Name)] = hook.Owner
-	}
-
-	for _, hook := range hooks.PreTerminate {
-		annotations[fmt.Sprintf("%s/%s", capiv1.PreTerminateDeleteHookAnnotationPrefix, hook.Name)] = hook.Owner
-	}
-
-	return annotations
-}
-
-// handleUnsupportedMachineFields checks for fields that are not supported by CAPI and returns a list of errors.
-func handleUnsupportedMachineFields(spec mapiv1.MachineSpec) field.ErrorList {
+// fromMAPIMachineToCAPIMachine translates a MAPI Machine to its Core CAPI Machine correspondent.
+func fromMAPIMachineToCAPIMachine(mapiMachine *mapiv1beta1.Machine, apiVersion, kind string) (*capiv1.Machine, field.ErrorList) {
 	var errs field.ErrorList
 
-	fldPath := field.NewPath("spec")
-
-	errs = append(errs, handleUnsupportedMAPIObjectMetaFields(fldPath.Child("metadata"), spec.ObjectMeta)...)
-
-	// TODO(OCPCLOUD-2861/2899): Taints are not supported by CAPI. add support for them via CAPI BootstrapConfig + minimal bootstrap controller.
-	if len(spec.Taints) > 0 {
-		errs = append(errs, field.Invalid(fldPath.Child("taints"), spec.Taints, "taints are not currently supported"))
+	capiMachine := &capiv1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            mapiMachine.Name,
+			Namespace:       capiNamespace,
+			Labels:          convertMAPILabelsToCAPI(mapiMachine.Labels),
+			Annotations:     convertMAPIAnnotationsToCAPI(mapiMachine.Annotations),
+			OwnerReferences: nil, // OwnerReferences not populated here. They are added later by the machineSync controller.
+			// Finalizers: // Finalizers not populated here. They are added later by the machine controllers.
```

Comment:
Why are these different? Why is one set to nil and the other just commented?

Reply:
Added in different iterations by different people, I'll switch to nil if it doesn't cause issues.